### PR TITLE
Add Leadfoot typings

### PR DESCRIPTION
**Pull request by rogierschouten**

### CODE REVIEW

**Comment by rogierschouten** [comment]
Note these typings were taken from https://github.com/theintern/intern/blob/master/typings/leadfoot/leadfoot.d.ts

The dojo typings were modified to make the leadfoot typings work.  Dojo is a client-side library but leadfoot uses a small part of it (its promises) on the server-side. This means that the typings for Dojo and Node must co-exist.  To this end I removed the globals from the Dojo typings.  I think in types-2.0, having globals is no longer the way to go anyway, right?

File: leadfoot/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+    "files": [
+        "index.d.ts",
+        "leadfoot-tests.ts",
+		"../node/index.d.ts",
```
Comment (mhegazy) [inline leadfoot/tsconfig.json]: do not include files outside the module. use `/// <reference types="node" />` instead

**Comment by mhegazy** [comment]
@vansimke can you review this change

**Review by mhegazy — CHANGES_REQUESTED** [review]
Please remove the `../` references form the tsconfig.json file

**Comment by rogierschouten** [comment]
@mhegazy done, thank you

**Review by mhegazy — APPROVED** [review]

**Pull request merged**